src/consensus/role_follower.rs

```diff
@@ -484,39 +524,36 @@ impl Consensus {
             return Ok(());
         }
 
-        std::mem::swap(
-            &mut self.bft_round_state.consensus_proposal,
-            &mut potential_proposal,
-        );
+        self.bft_round_state.current_proposal = potential_proposal;
         self.bft_round_state.joining.buffered_prepares.clear();
 
         info!(
             "📦 Commit message received for slot {}, trying to synchronize.",
-            self.bft_round_state.consensus_proposal.slot
+            self.bft_round_state.current_proposal.slot
         );
 
         // Try to commit the proposal
+        let old_slot = self.bft_round_state.slot;
+        let old_view = self.bft_round_state.view;
+
+        self.bft_round_state.slot = self.bft_round_state.current_proposal.slot;
+        self.bft_round_state.view = 0; // TODO
         self.bft_round_state.state_tag = StateTag::Follower;
         if self
             .try_commit_current_proposal(
                 commit_quorum_certificate,
-                self.bft_round_state.consensus_proposal.hashed(),
+                self.bft_round_state.current_proposal.hashed(),
             )
             .is_err()
         {
             // Swap back
-            std::mem::swap(
-                &mut self.bft_round_state.consensus_proposal,
-                &mut potential_proposal,
-            );
```

Comment:
remove comment too

Reply:
We swap back the slow/view so it's still accurate imo

Reply:
:indeed: